### DISCUSSION TITLE
Read a JWT header's value

### DISCUSSION THREAD

**Discussion by avishail** [lestrrat-go/jwx#573] [Q&A]
Edit: I was able to parse the JWT by hand according to this suggestion: https://github.com/lestrrat-go/jwx/discussions/547#discussioncomment-1961301

I just wondering if there is a library functionality to do that for us...

I was able to add a metadata to my JWT token's header using jwt.WithJwsHeaders(hdrs) option during singing. Now, I need this header's value **before** validating the token . This hint tells me what key version to use. I was trying to look in the examples and discussions about how to extract this header I added so I'll be able to verify the token but with no luck.

Any pointer to the right functionality will be welcomed!

Thanks,
Avishai

**Comment by lestrrat** [comment]
>  Now, I need this header's value before validating the token

I guess you already found your answer, but I guess this is something that this library may have to work on. Thanks for the feedback

**Reply by avishail** [reply]
Indeed. Thanks!
